lib/puppet/type/concat_file.rb

```diff
@@ -274,15 +271,13 @@ def nested_merge(hash1, hash2)
     return hash2 if hash1.nil? || hash1.empty?
 
     # Unique merge for arrays
-    if hash1.is_a?(Array) && hash2.is_a?(Array)
-      return (hash1 + hash2).uniq
-    end
+    return (hash1 + hash2).uniq if hash1.is_a?(Array) && hash2.is_a?(Array)
 
     # Deep-merge Hashes; higher order value is kept
     hash1.merge(hash2) do |k, v1, v2|
       if v1.is_a?(Hash) && v2.is_a?(Hash)
         nested_merge(v1, v2)
-      elsif v1.is_a?(Array) && v2.is_a?(Array)
+      elsif v1.is_a?(Array) && v2.is_a?(Array) # rubocop:disable Lint/DuplicateBranch
```

Comment:
Just a suggestion, totally up to you if ya want to implement this.
To get rid of the Lint/DuplicateBranch violation you can do this
```suggestion
      if (v1.is_a?(Hash) && v2.is_a?(Hash)) || (v1.is_a?(Array) && v2.is_a?(Array))
          nested_merge(v1, v2)
      else
```